aspnetcore/mvc/razor-pages/razor-pages-authorization-and-custom-routing.md

```diff
@@ -0,0 +1,71 @@
+
+## Require authorization to access a folder of pages
+
+Use the [AuthorizeFolder](/dotnet/api/microsoft.extensions.dependencyinjection.pageconventioncollectionextensions.authorizefolder) convention via [AddRazorPagesOptions](/dotnet/api/microsoft.extensions.dependencyinjection.mvcrazorpagesmvcbuilderextensions.addrazorpagesoptions) to add an [AuthorizeFilter](/dotnet/api/microsoft.aspnetcore.mvc.authorization.authorizefilter) to all of the pages in a folder at the specified path:
+
+[!code-csharp[Main](razor-pages-authorization-conventions/sample/Startup.cs?name=snippet1&highlight=2,5)]
+
+An [AuthorizeFolder overload](/dotnet/api/microsoft.extensions.dependencyinjection.pageconventioncollectionextensions.authorizefolder#Microsoft_Extensions_DependencyInjection_PageConventionCollectionExtensions_AuthorizeFolder_Microsoft_AspNetCore_Mvc_ApplicationModels_PageConventionCollection_System_String_System_String_) is available if you need to specify an authorization policy.
+
+## Allow anonymous access to a page
+
+Use the [AllowAnonymousToPage](/dotnet/api/microsoft.extensions.dependencyinjection.pageconventioncollectionextensions.allowanonymoustopage) convention via [AddRazorPagesOptions](/dotnet/api/microsoft.extensions.dependencyinjection.mvcrazorpagesmvcbuilderextensions.addrazorpagesoptions) to add an [AllowAnonymousFilter](/dotnet/api/microsoft.aspnetcore.mvc.authorization.allowanonymousfilter) to a page at the specified path:
+
+[!code-csharp[Main](razor-pages-authorization-conventions/sample/Startup.cs?name=snippet1&highlight=2,6)]
+
+## Allow anonymous access to a folder of pages
+
+Use the [AllowAnonymousToFolder](/dotnet/api/microsoft.extensions.dependencyinjection.pageconventioncollectionextensions.allowanonymoustofolder) convention via [AddRazorPagesOptions](/dotnet/api/microsoft.extensions.dependencyinjection.mvcrazorpagesmvcbuilderextensions.addrazorpagesoptions) to add an [AllowAnonymousFilter](/dotnet/api/microsoft.aspnetcore.mvc.authorization.allowanonymousfilter) to all of the pages in a folder at the specified path:
+
+[!code-csharp[Main](razor-pages-authorization-conventions/sample/Startup.cs?name=snippet1&highlight=2,7)]
+
+## Note on combining authorized and anonymous access
+
+It's perfectly valid to specify that a folder of pages require authorization and specify that a page within that folder allows anonymous access:
+
+```csharp
+// This works.
+.AuthorizeFolder("/Private").AllowAnonymousToPage("/Public")
+```
+
+The reverse, however, isn't true. You can't declare a folder of pages for anonymous access and specify a page within for authorization:
+
+```csharp
+// This doesn't work!
+.AllowAnonymousToFolder("/Public").AuthorizePage("/Private") 
```

Comment:
`.AllowAnonymousToFolder("/Public").AuthorizePage("/Public/Private") ` since it's a page inside the directory where the rules break down.